gcp_variant_transforms/testing/integration/run_tests_common.py

```diff
@@ -16,60 +16,65 @@
 
 from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
 
-DEFAULT_IMAGE_NAME = 'gcr.io/gcp-variant-transforms/gcp-variant-transforms'
+_DEFAULT_IMAGE_NAME = 'gcr.io/gcp-variant-transforms/gcp-variant-transforms'
+_DEFAULT_ZONES = ['us-west1-b']
 
 
 class TestCaseFailure(Exception):
   """Exception for failed test cases."""
   pass
 
 
-class TestCase(object):
-  """Test case that holds information to run in pipelines API."""
-
-  def __init__(self, test_name, project):
-    # type: (str, str) -> None
-    self._name = test_name
-    self._project = project
-
-  def get_name(self):
-    return self._name
-
-  def run(self, context):
-    service = discovery.build(
-        'genomics', 'v1alpha2', credentials=context.credentials)
-    # The following pylint hint is needed because `pipelines` is a method that
-    # is dynamically added to the returned `service` object above. See
-    # `googleapiclient.discovery.Resource._set_service_methods`.
-    # pylint: disable=no-member
-    request = service.pipelines().run(body=self._pipelines_api_request)
-
-    operation_name = request.execute()['name']
-    response = self._wait_for_operation_done(service, operation_name)
-    self._handle_failure(response)
-
-  def _wait_for_operation_done(self, service, operation_name):
-    """Waits until the operation ``operation_name`` of ``service`` is done."""
+class TestRunner(object):
+  """Runs the tests using pipelines API."""
+
+  def __init__(self, tests):
+    # type: (List[Union[PreprocessorTestCase, VcfToBQTestCase]]) -> None
+    self._tests = tests
+    self._service = discovery.build(
+        'genomics',
+        'v1alpha2',
+        credentials=GoogleCredentials.get_application_default())
+    self._operation_names = []
+    self._responses = []
+
+  def run(self):
+    """Runs all tests."""
+    for test in self._tests:
+      # The following pylint hint is needed because `pipelines` is a method that
+      # is dynamically added to the returned `service` object above. See
+      # `googleapiclient.discovery.Resource._set_service_methods`.
+      # pylint: disable=no-member
+      request = self._service.pipelines().run(body=test.pipeline_api_request)
+      operation_name = request.execute()['name']
+      self._operation_names.append(operation_name)
+    self._wait_for_all_operations_done()
+
+  def _wait_for_all_operations_done(self):
+    """Waits until all operations of ``service`` are done."""
     time.sleep(60)
-    operations = service.operations()
-    request = operations.get(name=operation_name)
-    response = request.execute()
-    while not response['done']:
-      time.sleep(10)
+    # pylint: disable=no-member
+    operations = self._service.operations()
+    for operation_name in self._operation_names:
+      request = operations.get(name=operation_name)
       response = request.execute()
-    return response
+      while not response['done']:
```

Comment:
Suggestion: Instead of waiting for a each operation to be 'done' until checking the next one, I think it makes more sense to check all operations in a round-robin fashion and whenever one is 'done' do the `_handle_failure`, i.e., change this `while` to an `if response['done']:` and change the outer `for` to a while that works until there are no `_operation_names` that is not 'done'. The benefit of that approach is that if any of the tests fail early, we will fail the whole process early and not waiting for a long test to finish until checking that failed test.